pkgs/by-name/af/affine/package.nix

```
@@ -35,212 +36,208 @@ let
-          zip
-          jq
-          rsync
-        ]
-        ++ lib.optionals hostPlatform.isLinux [
-          copyDesktopItems
-          makeWrapper
-        ];
-
-      patchPhase = ''
-        runHook prePatchPhase
-
-        sed -i '/packagerConfig/a \    electronZipDir: process.env.ELECTRON_FORGE_ELECTRON_ZIP_DIR,' packages/frontend/apps/electron/forge.config.mjs
-
-        runHook postPatchPhase
+stdenv.mkDerivation (finalAttrs: {
+  productName = if buildType == "stable" then "AFFiNE" else "AFFiNE-" + buildType;
+  binName = lib.toLower finalAttrs.productName;
+  pname = finalAttrs.binName;
+
+  # https://github.com/toeverything/AFFiNE/releases/tag/v0.18.1
+  version = "0.19.6";
+  src = fetchFromGitHub {
+    owner = "toeverything";
+    repo = "AFFiNE";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-BydTNE36oRIxr2lTnc2+EY0lvMXn4NTLB4EjqzhdjGk=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    src = finalAttrs.src;
+    hash = "sha256-racjpf0VgNod6OxWKSaCbKS9fEkInpDyhVbAHfYWIDo=";
+  };
+  yarnOfflineCache = stdenvNoCC.mkDerivation {
+    name = "yarn-offline-cache";
+    src = finalAttrs.src;
+    nativeBuildInputs = [
+      yarn
+      cacert
+      writableTmpDirAsHomeHook
+    ];
+    supportedArchitectures = builtins.toJSON {
+      os = [
+        "darwin"
+        "linux"
+      ];
+      cpu = [
+        "arm64"
+        "x64"
+      ];
+      libc = [
+        "glibc"
+        "musl"
+      ];
+    };
+    buildPhase = ''
+      runHook preBuild
+
+      export CI=1
+
+      mkdir -p $out
+      yarn config set enableTelemetry false
+      yarn config set cacheFolder $out
+      yarn config set enableGlobalCache false
+      yarn config set supportedArchitectures --json "$supportedArchitectures"
+
+      yarn install --immutable --mode=skip-build
+
+      runHook postBuild
+    '';
+    dontInstall = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-E9l5zjOOfyDBzYJOU94VrRvt7Hi4XkRTDav9bVlXvlQ=";
+  };
+  nativeBuildInputs =
+    [
+      nodejs
+      yarn
+      cargo
+      rustc
+      findutils
+      zip
+      jq
+      rsync
+      writableTmpDirAsHomeHook
+    ]
+    ++ lib.optionals hostPlatform.isLinux [
+      copyDesktopItems
+      makeWrapper
+    ];
+
+  # Remove code under The AFFiNE Enterprise Edition (EE) license.
+  # Keep file package.json for `yarn install --immutable` lockfile check.
+  postPatch = ''
+    BACKEND_SERVER_PACKAGE_JSON="$(jq 'del(.scripts.postinstall)' packages/backend/server/package.json)"
+    rm -rf packages/backend/server/{.*,*}
+    echo "$BACKEND_SERVER_PACKAGE_JSON" > packages/backend/server/package.json
+  '';
+  configurePhase =
+    let
+      electronContentPath =
+        electron + (if hostPlatform.isLinux then "/libexec/electron/" else "/Applications/");
+    in
+    ''
+      runHook preConfigurePhase
+
+      export CI=1
+
+      # cargo config
+      mkdir -p .cargo
+      cat $cargoDeps/.cargo/config.toml >> .cargo/config.toml
```

```suggestion
      cat $cargoDeps/.cargo/config.toml >> ~/.cargo/config.toml
```
Previously $HOME was the build directory

---

I think `.cargo/config.toml` should be in `$NIX_BUILD_TOP/source` dir, which is not `$NIX_BUILD_TOP/.home` dir defined in writableTmpDirAsHomeHook.

---

I'm not sure. What I do know is that cargo will certainly read `~/.cargo/config.toml` for all cargo invocations, so it seems the safer option. Maybe either way works.

---

``` 
error: failed to get `chrono` as a dependency of package `affine_server_native v1.0.0 (/private/tmp/nix-build-affine-0.19.6.drv-0/source/packages/backend/native)`

Caused by:
  failed to load source for dependency `chrono`
```

So I can't change this to ~/.cargo/config.toml